be/src/io/fs/s3_connectivity_checker.cpp

```diff
@@ -0,0 +1,139 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "io/fs/s3_connectivity_checker.h"
+
+#include <aws/core/auth/AWSCredentials.h>
+#include <aws/s3/S3Client.h>
+#include <aws/s3/model/HeadBucketRequest.h>
+
+#include "common/logging.h"
+#include "util/s3_uri.h"
+#include "util/s3_util.h"
+
+namespace doris::io {
+
+S3ConnectivityChecker::S3ConnectivityChecker(const std::map<std::string, std::string>& properties) {
+    auto status = _parse_properties(properties);
+    if (!status.ok()) {
+        LOG(WARNING) << "Failed to parse S3 properties: " << status.to_string();
+    }
+}
+
+Status S3ConnectivityChecker::check() {
+    // Validate test_location
+    if (_test_location.empty()) {
+        return Status::InvalidArgument("Missing 'test_location' property");
+    }
+
+    // Extract bucket from test_location
+    std::string bucket;
+    RETURN_IF_ERROR(_extract_bucket_from_location(_test_location, &bucket));
+
+    // Validate required properties for creating S3 client
+    if (_endpoint.empty()) {
+        return Status::InvalidArgument("Missing 'AWS_ENDPOINT' property");
+    }
+    if (_region.empty()) {
+        return Status::InvalidArgument("Missing 'AWS_REGION' property");
+    }
+    if (_access_key.empty()) {
+        return Status::InvalidArgument("Missing 'AWS_ACCESS_KEY' property");
+    }
+    if (_secret_key.empty()) {
+        return Status::InvalidArgument("Missing 'AWS_SECRET_KEY' property");
+    }
+
+    // Configure AWS client
+    Aws::Client::ClientConfiguration config = S3ClientFactory::getClientConfiguration();
+    config.endpointOverride = _endpoint;
+    config.region = _region;
+
+    // Create credentials
+    Aws::Auth::AWSCredentials credentials(_access_key, _secret_key);
```

Comment:
I think the builder of `Aws::S3::S3Client` should be wrapped in `be/src/util/s3_util.cpp`, so that all callers share the same code